src/api/clock.rs

```diff
@@ -0,0 +1,52 @@
+use embedded_time::TimeInt;
+use embedded_time::fraction::Fraction;
+
+#[derive(Clone, Copy, PartialOrd, Ord, Eq, fmt::Debug)]
```

Comment:
I'm surprised you can derive `Eq` without `PartialEq`. You should add `PartialEq`.

Reply:
Yes, my mistake.